### PR TITLE
Adds a RPI Pico Example: Azure_IoT_Hub_RPI_Pico

### PR DESCRIPTION
**Pull request by djaus2**
Based upon the ESP8266 example.
Only requires a few lines of code changes from that.

Note: Main branch readme for this project has code changes at bottom.

Nb: This replaces #49

Note: Also fixes warning you get in ```getTelemetryPayload( )```

### CODE REVIEW

File: examples/Azure_IoT_Hub_RPI_Pico/readme.md [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+    -   [Troubleshooting](#troubleshooting)
+
+    -   [Contributing](#contributing)
+
+        -   [License](#license)
+
+
+## How to install Arduino BSP onto a RPI Pico
+- I was given a [Freenove Basic Pico Starter Kit](https://github.com/Freenove/Freenove_Basic_Starter_Kit_for_Raspberry_Pi_Pico) and a [RPI Pico W](https://www.raspberrypi.com/documentation/microcontrollers/raspberry-pi-pico.html#raspberry-pi-pico-w) for Xmas!!
```
Comment (CIPop) [inline examples/Azure_IoT_Hub_RPI_Pico/readme.md:41]: Please remove the starter kit reference, etc (things not relevant and potentially confusing for the sample).

File: examples/Azure_IoT_Hub_RPI_Pico/readme.md [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+- Ref: [https://freenove.com/fnk0058/](https://freenove.com/fnk0058/)
```
Comment (CIPop) [inline examples/Azure_IoT_Hub_RPI_Pico/readme.md:42]: Please remove: let's keep instructions simple.

File: examples/Azure_IoT_Hub_RPI_Pico/readme.md [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+  - The repository [earlephilhower-arduino-pico on Github](https://github.com/earlephilhower/arduino-pico)  
+  - Thx Earle.
+- Open Arduino IDE, and click File>Preference. In new pop-up window, find “Additional Boards Manager 
+URLs”, and replace with a new line：
+```
+https://github.com/earlephilhower/arduino-pico/releases/download/global/package_rp2040_index.json
+```
+- In the IDE click Tools>Board>Boards Manager...on the menu bar
+- Enter Pico in the searching box, and select ”Raspberry Pi Pico/RP2040” and click on [Install]
```
Comment (CIPop) [inline examples/Azure_IoT_Hub_RPI_Pico/readme.md:51]: This is an unofficial package. Would using the official Arduino Mbed OS RP2040 Boards by Arduino work or does that not have WiFi support yet?

File: examples/Azure_IoT_Hub_RPI_Pico/readme.md [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+- Click Yes in the pop-up “dpinst-amd64.exe”installation window.
```
Comment (CIPop) [inline examples/Azure_IoT_Hub_RPI_Pico/readme.md:52]: Please add the version this was last tested with (e.g. v2.7.0)

Comment (CIPop) [inline examples/Azure_IoT_Hub_RPI_Pico/readme.md:52]: I did not get a pop-up for v2.7.0

File: examples/Azure_IoT_Hub_RPI_Pico/readme.md [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+- Starting with the Pico disconnected from computer: Press and keep pressing the white button (BOOTSEL) on Pico, and connect Pico to 
+computer using the USB port before releasing the button. (Note: Be sure to keep pressing the button before powering the Pico, 
+otherwise the firmware will not download successfully)/
+- In the Arduino IDE. Click File>Examples>01.Basics>Blink.
+- Click Tools>Board>Raspberry Pi RP2040 Boards>Raspberry Pi Pico.
```
Comment (CIPop) [inline examples/Azure_IoT_Hub_RPI_Pico/readme.md:57]: Shouldn't this be Pi Pico __W__?

File: examples/Azure_IoT_Hub_RPI_Pico/readme.md [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+    [here](https://docs.microsoft.com/azure/iot-pnp/howto-use-iot-explorer).
+
+    NOTE: This guide demonstrates use of the Azure CLI and does NOT demonstrate
+    use of Azure IoT Explorer.
+
+## Setup and Run Instructions
+
+1.  Run the Arduino IDE.
+
+2.  Install the Azure SDK for Embedded C library.
```
Comment (CIPop) [inline examples/Azure_IoT_Hub_RPI_Pico/readme.md:145]: Azure SDK for C

File: examples/Azure_IoT_Hub_RPI_Pico/readme.md [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+
+3.  Install the Arduino PubSubClient library. (PubSubClient is a popular MQTT
+    client for Arduino.)
+
+    -   On the Arduino IDE, go to menu `Sketch`, `Include Library`, `Manage
+        Libraries...`.
+
+    -   Search for `PubSubClient` (by Nick O'Leary).
+
+    -   Hover over the library item on the result list, then click on "Install".
```
Comment (CIPop) [inline examples/Azure_IoT_Hub_RPI_Pico/readme.md:160]: Please include the version v2.8.0.

File: examples/Azure_IoT_Hub_RPI_Pico/readme.md [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+
+    -   Search for `TimeLib` aka Time (by Michael Margiolis).
+ 
+    -   Hover over the library item on the result list, then click on "Install".
+
+4.  Open the sample.
+
+    -   On the Arduino IDE, go to menu `File`, `Examples`, `azure-sdk-for-c`.
+
+    -   Click on `azure_esp8266` to open the sample. <-- This needs to change when this is fomally merged into the original repository.
```
Comment (CIPop) [inline examples/Azure_IoT_Hub_RPI_Pico/readme.md:172]: Please update

File: examples/Azure_IoT_Hub_RPI_Pico/readme.md [hunk 9]
```diff
@@ -0,0 +1,318 @@
...
+    
+    -   **For now** Download the repository and open the Sketch in this folder, in the IDE. 
+        ```azure-sdk-for-c-arduino\examples\Azure_IoT_Hub_RPI_Pico```
+
+      -  **Alt** Open the azure esp8266 example as above and make the changes as [here]().
```
Comment (CIPop) [inline examples/Azure_IoT_Hub_RPI_Pico/readme.md:177]: Please remove.

File: examples/Azure_IoT_Hub_RPI_Pico/Azure_IoT_Hub_RPI_Pico.ino [hunk 10]
```diff
@@ -0,0 +1,366 @@
...
+
+// Additional sample headers
+#include "iot_configs.h"
+
+// When developing for your own Arduino-based platform,
+// please follow the format '(ard;<platform>)'. (Modified this)
+#define AZURE_SDK_CLIENT_USER_AGENT "c%2F" AZ_SDK_VERSION_STRING "(ard;rpipico)"
+
+// Utility macros and defines
+#define LED_PIN 2
```
Comment (CIPop) [inline examples/Azure_IoT_Hub_RPI_Pico/Azure_IoT_Hub_RPI_Pico.ino:57]: Use LED_BUILTIN instead. Current configuration won't work on a RP2040 board.

**Review by CIPop — CHANGES_REQUESTED** [review]
Thank you for your contribution @djaus2!
We cannot offer support for this new sample since it is not using official RPI or Arduino SDKs.

Our recommendation is to create a separate repo within your account. Once that is set-up, we will add a link to our main `README.md` file to point to this community-supported, unofficial sample.

File: examples/Azure_IoT_Hub_RPI_Pico/readme.md (hunk 3 above)
Comment (djaus2) [reply examples/Azure_IoT_Hub_RPI_Pico/readme.md:51]: Doesn't have WiFi

File: examples/Azure_IoT_Hub_RPI_Pico/readme.md (hunk 4 above)
Comment (djaus2) [reply examples/Azure_IoT_Hub_RPI_Pico/readme.md:52]: IDE V2.03

**Comment by CIPop** [comment]
Closing PR with the following recommendation:

> Our recommendation is to create a separate repo within your account. Once that is set-up, we will add a link to our main README.md file to point to this community-supported, unofficial sample.

**Comment by djaus2** [comment]
I'll make those changes but I also have been working on a signifcant port of the complete (as much is possible) of the Azure SDK for Embedded C functionality so when that is complete I'll link that.

**Pull request closed without merging**